### PR TITLE
Xenial docs

### PR DESCRIPTION
**Pull request by joepvd**

### CODE REVIEW

File: user/reference/xenial.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+Please note that this image will only be available on the sudo-enabled
+infrastructure. If you are running an Enterprise installation, you can use the
+Xenial Docker images.
+
+## Image differences from Trusty
+Apart from version bumps of software, and new defaults for language
+environment, the Xenial image contains the following changes:
+
+### Non-core apt-repositories removed
+On Trusty, apt repositories that have been used to install images have been
```
Comment (BanzaiMan) [inline user/reference/xenial.md]: "install images" → "install packages"

File: user/reference/xenial.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+| mongodb       | "deb http://repo.mongodb.org/apt/ubuntu xenial/mogodb-org/4.0 multiverse" |
+| couchdb       | ppa:couchdb/stable         |
+| docker        | "deb [arch=amd64] https://download.docker.com/linux/ubuntu xenial stable" |
+| git-lfs       | "deb https://packagecloud.io/github/git-lfs/ubuntu/ xenial main" |
+| git-ppa       | ppa:git-core/ppa           |
+| google_chrome | "deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main" |
+| pollinate     | ppa:pollinate/ppa          |
+
+### Services disabled by default
+On Trusty, some services that you might or might not use, were running by
```
Comment (BanzaiMan) [inline user/reference/xenial.md]: "were" → "are" We still support Trusty, so let's use the present tense.

File: user/reference/xenial.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+## Using Xenial
+The following configuration will get your builds to run on a Xenial image:
+
+```yaml
+dist: xenial
+```
+{: data-file=".travis.yml"}
+
+Please note that this image will only be available on the sudo-enabled
```
Comment (MariadeAnton) [inline user/reference/xenial.md]:
```suggestion
Please note that Xenial is available on our fully virtualized
```

File: user/reference/xenial.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+infrastructure. If you are running an Enterprise installation, you can use the
```
Comment (MariadeAnton) [inline user/reference/xenial.md]:
```suggestion
infrastructure. If you are running an Enterprise installation, please reach out
```

File: user/reference/xenial.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+Xenial Docker images.
```
Comment (MariadeAnton) [inline user/reference/xenial.md]:
```suggestion
to [enterprise@travis-ci.com](mailto:entereprise@travis-ci.com) to see how you can use the Xenial Docker images.
```

File: user/reference/xenial.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+## Image differences from Trusty
+Apart from version bumps of software, and new defaults for language
```
Comment (MariadeAnton) [inline user/reference/xenial.md]:
```suggestion
Xenial includes the following changes and improvements: 
```

File: user/reference/xenial.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+environment, the Xenial image contains the following changes:
```
Comment (MariadeAnton) [inline user/reference/xenial.md]:
```suggestion

```

File: user/reference/xenial.md [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+
+### Non-core apt-repositories removed
```
Comment (MariadeAnton) [inline user/reference/xenial.md]:
```suggestion
### Third party apt-repositories removed
```

File: user/reference/xenial.md (hunk 1 above)
Comment (MariadeAnton) [inline user/reference/xenial.md]:
```suggestion
On Trusty, apt repositories that have been used to install packages have been
```

File: user/reference/xenial.md [hunk 9]
```diff
@@ -0,0 +1,153 @@
+---
+title: The Xenial Build Environment
+layout: en
+---
+
+## What This Guide Covers
+
+On this page, you can find what is included on the Xenial build environment.
```
Comment (plaindocs) [inline user/reference/xenial.md]:
```suggestion
This guide provides an overview of the packages, tools and settings available in the Xenial environment.
```

File: user/reference/xenial.md [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+
+## Using Xenial
+
+To get your builds using Ubuntu Xenial 16.04, add the following to your `.travis.yml`:
```
Comment (plaindocs) [inline user/reference/xenial.md]:
```suggestion
To route your builds to Ubuntu Xenial 16.04, add the following to your `.travis.yml`:
```

File: user/reference/xenial.md [hunk 11]
```diff
@@ -0,0 +1,153 @@
...
+
+```yaml
+dist: xenial
+```
+{: data-file=".travis.yml"}
+
+Please note that Xenial is available on our fully virtualized
+infrastructure. If you are running an Enterprise installation, please reach out
+to [enterprise@travis-ci.com](mailto:entereprise@travis-ci.com) to see how you can use the Xenial Docker images.
```
Comment (plaindocs) [inline user/reference/xenial.md]: We should probably add a subject to this?

File: user/reference/xenial.md [hunk 12]
```diff
@@ -0,0 +1,153 @@
...
+
+To get your builds using Ubuntu Xenial 16.04, add the following to your `.travis.yml`:
+
+```yaml
+dist: xenial
+```
+{: data-file=".travis.yml"}
+
+Please note that Xenial is available on our fully virtualized
+infrastructure. If you are running an Enterprise installation, please reach out
```
Comment (plaindocs) [inline user/reference/xenial.md]: As that is now (ish) our only, do we can to say "hosted" or something?

File: user/reference/xenial.md [hunk 13]
```diff
@@ -0,0 +1,153 @@
...
+to [enterprise@travis-ci.com](mailto:entereprise@travis-ci.com) to see how you can use the Xenial Docker images.
+
+## Image updates from Trusty
```
Comment (plaindocs) [inline user/reference/xenial.md]:
```suggestion
## Differences from our Trusty Infrastructure
```

File: user/reference/xenial.md [hunk 14]
```diff
@@ -0,0 +1,153 @@
...
+
+Xenial includes the following changes and improvements:
+
+### Third party apt-repositories removed
+
+While third party apt-repositories are used during the Xenial image provisioning, they are all removed from the Xenial build image. This has two benefits; a) reduced risk of unrelated interference and b) faster apt-get updates.
+
+To specify a third party apt-repository, you can [add the source with the apt addon](/user/installing-dependencies/#adding-apt-sources) and specify the packages. For example:
```
Comment (plaindocs) [inline user/reference/xenial.md]: Do you need to add both? I think not. You could add just the source? Or the source and the package? We could maybe head off some confusion here.

File: user/reference/xenial.md [hunk 15]
```diff
@@ -0,0 +1,153 @@
...
+| docker        | `"deb [arch=amd64] https://download.docker.com/linux/ubuntu xenial stable"` |
+| git-lfs       | `"deb https://packagecloud.io/github/git-lfs/ubuntu/ xenial main"`          |
+| git-ppa       | `ppa:git-core/ppa`                                                          |
+| google_chrome | `"deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main"`     |
+| pollinate     | `ppa:pollinate/ppa`                                                         |
+
+### Services disabled by default
+
+On Trusty, some services that you might or might not use, were running by
+default. On Xenial, to optimize the use of the virtual machine resources, they come disabled by default.
```
Comment (plaindocs) [inline user/reference/xenial.md]: What about replacing all of this with

```suggestion
On our Xenial infrastructure, to speed up boot time and improve performance we've disabled all services, including the ones that were started by default on Trusty. Add any services that you want to start by default to your `.travis.yml`:

File: user/reference/xenial.md [hunk 16]
```diff
@@ -0,0 +1,153 @@
...
+```yaml
+services:
+  - mysql
+  - redis
+```
+{: data-file=".travis.yml"}
+
+## Environment common to all Xenial images
+
+In total, three different Xenial images are available for use. The selection
```
Comment (plaindocs) [inline user/reference/xenial.md]: These paragraphs are well confusing, and we don't mention `steveonnie` anywyere else. I'd b tempted to either delete all of these paragraphs up to the common environment, or copy some of text from Ubuntu about Minimal and Generic.

WDYT?

File: user/reference/xenial.md [hunk 17]
```diff
@@ -0,0 +1,153 @@
...
+repositories.
+
+### Docker
+
+* Docker 18.06.0-ce is installed
+* docker-compose 1.23.1.
+
+## Ruby images
+
+* Pre-installed Rubies: `2.4.5` and `2.5.3`.
```
Comment (plaindocs) [inline user/reference/xenial.md]: What we could do is add a line for each language here about what base image is uses?

File: user/reference/xenial.md [hunk 18]
```diff
@@ -0,0 +1,153 @@
...
+
+## JavaScript and Node.js images
+
+* The newest nvm is available to all images. (FIX: is there a version we can use here?)
+
+* Pre-installed NodeJS versions: `11.0.0` and `8.12.0`.
+
+## Go images
+
+* * Pre-installed Go: `1.11.1`
```
Comment (plaindocs) [inline user/reference/xenial.md]:
```suggestion
* Pre-installed Go: `1.11.1`
```

File: user/reference/xenial.md [hunk 19]
```diff
@@ -0,0 +1,153 @@
...
+
+* * Other ruby versions can be installed during build time by specifying the language versions with the `go:`-key.
```
Comment (plaindocs) [inline user/reference/xenial.md]:
```suggestion
* Other ruby versions can be installed during build time by specifying the language versions with the `go:`-key.
```

File: user/reference/xenial.md [hunk 20]
```diff
@@ -0,0 +1,153 @@
...
+* The following table summarizes the Pre-installed JVM tooling versions:
+
+| package | version |
+|:--------|:--------|
+| gradle  | 4.10.2  |
+| maven   | 3.5.4   |
+
+## Databases and services
+
+* Pre-installed Databases and services:
```
Comment (plaindocs) [inline user/reference/xenial.md]:
```suggestion
Pre-installed Databases and services:
```

Comment (plaindocs) [reply user/reference/xenial.md]: And maybe a reminder here that you need to start them?

File: user/reference/xenial.md [hunk 21]
```diff
@@ -99,36 +93,40 @@ You can find below the environment common to all Xenial images. The environment
...
 
 * Python `2.7.15` will be used when no language version is explicitly set.
 
-## JavaScript and Node.js images
-
-* Nvm is automatically updated to the latest version at build time. 
+## JavaScript and Node.js support
 
-* Pre-installed NodeJS versions: `11.0.0` and `8.12.0`.
+* For builds specifying `language: node_js`, `nvm` is automatically updated to the latest version at build time. For other builds, the stable version at image build time has been selected, which is 0.33.11.
```
Comment (plaindocs) [inline user/reference/xenial.md]: :+1:

File: user/reference/xenial.md [hunk 22]
```diff
@@ -74,15 +74,9 @@ services:
...
-In total, three different Xenial images are available for use. The selection
-process takes place based on the language and services specified in your `travis.yml` configuration.
-
-The minimal image `stevonnie` supports docker, bash, and has the tools for dynamic language runtime download and activation, for example, `rvm` for ruby.
-
-The other two images have different language runtimes installed by default, and
-have an array of different databases and services available.
-
-You can find below the environment common to all Xenial images. The environment specifications below are also the ones for our `language: minimal` Xenial image.
+As a technicallity, three different images have been created that all support a
```
Comment (plaindocs) [inline user/reference/xenial.md]: This has the same issue of mentioning images but not really explaining them. What do you think of:

The following versions of Docker, version control software and compilers are present on all builds, along with more language specific software described in more detail in each language section.

File: user/reference/xenial.md [hunk 23]
```diff
@@ -141,13 +139,30 @@ repositories.
...
+| alias | version |
+|:------|:--------|
+| 5.6   | 5.6.36  |
+| 7.1   | 7.1.19  |
+| 7.2   | 7.2.7   |
+
 ## Databases and services
 
-Pre-installed Databases and services:
+Pre-installed Databases and services are:
```
Comment (plaindocs) [inline user/reference/xenial.md]: How about being more explicit here.

The following services and databases are preinstalled but but do not run by default. To use any of them, add it to the `services` key in your `travis.yml`:

Which also gets rid o ftext before _and_ after the table

File: user/database-setup.md [hunk 24]
```diff
@@ -58,10 +58,10 @@ and a blank password.
 Current versions of MySQL are
 
 
-|                 | Ubuntu Precise | Ubuntu Trusty |
-|:----------------|:---------------|:--------------|
-| Sudo-enabled    | 5.5.x          | 5.6.x         |
-| Container-based | -              | 5.6.x         |
+|                 | Ubuntu Precise | Ubuntu Trusty | Ubuntu Xenial |
```
Comment (plaindocs) [inline user/database-setup.md]: :+1:

File: user/database-setup.md [hunk 25]
```diff
@@ -106,7 +106,7 @@ before_install:
 
 ### MySQL 5.7
 
-On *sudo-enabled* Trusty Linux, you can install MySQL 5.7 by adding the following lines to your `.travis.yml`:
+MySQL 5.7 is the default on the Xenial image. On *sudo-enabled* Trusty Linux, you can install MySQL 5.7 by adding the following lines to your `.travis.yml`:
```
Comment (plaindocs) [inline user/database-setup.md]: :+1:

**Comment by plaindocs** [comment]
OK, I've resolved most of my own comments, and I'm happy for anyone to :shipit:

Great work @joepvd and @MariadeAnton !

**Pull request merged**